forest/cli/src/cli/snapshot_cmd.rs

```diff
@@ -0,0 +1,325 @@
+
+                let vars = HashMap::from([
+                    ("year".to_string(), year.to_string()),
+                    ("month".to_string(), month_string),
+                    ("day".to_string(), day_string),
+                    ("chain".to_string(), chain_name),
+                    ("height".to_string(), epoch.to_string()),
+                ]);
+                let output_path = match strfmt(&output_path.display().to_string(), &vars) {
+                    Ok(path) => path.into(),
+                    Err(e) => {
+                        cli_error_and_die(format!("Unparsable string error: {}", e), 1);
+                    }
+                };
+
+                let params = (
+                    epoch,
+                    *recent_stateroots,
+                    *include_old_messages,
+                    output_path,
+                    TipsetKeysJson(chain_head.key().clone()),
+                    *skip_checksum,
+                );
+
+                // infallible unwrap
+                let out = chain_export(params).await.map_err(handle_rpc_err).unwrap();
+
+                println!("Export completed. Snapshot located at {}", out.display());
+            }
+            Self::Fetch { snapshot_dir } => {
+                let snapshot_dir = snapshot_dir
+                    .clone()
+                    .unwrap_or_else(|| default_snapshot_dir(&config));
+                match snapshot_fetch(&snapshot_dir, config).await {
+                    Ok(out) => println!("Snapshot successfully downloaded at {}", out.display()),
+                    Err(e) => cli_error_and_die(format!("Failed fetching the snapshot: {e}"), 1),
+                }
+            }
+            Self::Dir => {
+                let dir = default_snapshot_dir(&config);
+                println!("Default snapshot dir: {}", dir.display());
+            }
+            Self::List { snapshot_dir } => {
+                list(&config, snapshot_dir);
+            }
+            Self::Remove {
+                filename,
+                snapshot_dir,
+                force,
+            } => {
+                remove(&config, filename, snapshot_dir, *force);
+            }
+            Self::Prune {
+                snapshot_dir,
+                force,
+            } => {
+                prune(&config, snapshot_dir, *force);
+            }
+        }
+    }
+}
+
+fn list(config: &Config, snapshot_dir: &Option<PathBuf>) {
+    let snapshot_dir = snapshot_dir
+        .clone()
+        .unwrap_or_else(|| default_snapshot_dir(config));
+    println!("Snapshot dir: {}", snapshot_dir.display());
+    if let Ok(dir) = fs::read_dir(snapshot_dir) {
+        println!("\nLocal snapshots:");
+        dir.flatten()
+            .map(|entry| entry.path())
+            .filter(|p| {
+                p.is_file()
+                    && p.extension()
+                        .map(|ext| ext.to_str().unwrap_or_default() == "car")
+                        .unwrap_or_default()
```

Comment:
do we need this map or can we simplify this?
```suggestion
                    && p.extension().unwrap_or_default() == "car"
```

Reply:
probably `is_file` is not needed, I guess for directories the `extension` then would return `None` which would not match witch `car`.

Reply:
Fixed. Didn't know `&OsStr == &str` works